content/recipes/swc.md

```diff
@@ -95,3 +95,100 @@ If your ORM does not provide a similar workaround, you can define the wrapper ty
  */
 export type WrapperType<T> = T; // WrapperType === Relation
 ```
+
+### Vitest
+
+[Vitest](https://vitest.dev/) is a fast and lightweight test runner designed to work with Vite. It provides a modern, fast, and easy-to-use testing solution that can be integrated with NestJS projects.
+
+#### Installation
+
+To get started, first install the required packages:
+
+```bash
+$ npm i --save-dev vitest unplugin-swc @swc/core @vitest/coverage-c8
+```
+
+#### Configuration
+
+Create a `vitest.config.ts` file in the root directory of your application with the following content:
+
+```ts
+import swc from 'unplugin-swc';
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+  test: {
+    globals: true,
+    root: './',
+  },
+
+  plugins: [swc.vite()], // This is required to build the test files with SWC
+});
+```
+
+This configuration file sets up the Vitest environment with the necessary aliases, root directory, and SWC plugin. You should also create a separate configuration
+file for e2e tests, with an additional field `include` that specifies the test path regex:
+
+```ts
+import swc from 'unplugin-swc';
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+  test: {
+    include: ['**/*.e2e-spec.ts'],
+    globals: true,
+    root: './',
+  },
+  plugins: [swc.vite()],
+});
+```
+
+Additionaly, you can set the `alias` options to support typescript paths in your tests:
+
+```ts
+import swc from 'unplugin-swc';
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+  test: {
+    include: ['**/*.e2e-spec.ts'],
+    globals: true,
+    alias: {
+      '@src': './src',
+      '@test': './test',
+    },
+    root: './',
+  },
+  resolve: {
+    alias: {
+      '@src': './src',
+      '@test': './test',
+    },
+  },
+  plugins: [swc.vite()],
+});
+```
+
+#### Update imports in E2E tests
+
+Change any E2E test imports using `import * as request from 'supertest'` to `import request from 'supertest';`. This is necessary because Vitest, when bundled with Vite, expects a default import for supertest. Using a namespace import may cause issues in this specific setup.
```

Comment:
```suggestion
Change any E2E test imports using `import * as request from 'supertest'` to `import request from 'supertest'`. This is necessary because Vitest, when bundled with Vite, expects a default import for supertest. Using a namespace import may cause issues in this specific setup.
```